flang/lib/Lower/Bridge.cpp

```diff
@@ -243,6 +243,11 @@ class FirConverter : public Fortran::lower::AbstractConverter {
     return lookupSymbol(sym).getAddr();
   }
 
+  void setSymbolAddress(Fortran::lower::SymbolRef sym,
+                        mlir::Value val) override final {
+    addSymbol(sym, val, /*forced=*/true);
```

Comment:
Yes, I see that it results in an error. And it works using the pushScope and popScope in `genFIR(const Fortran::parser::OpenMPConstruct)`.
But I am a bit confused by what should be the definition of scope here.
1) Any OpenMP construct containing a region should push a new scope.
2) Any construct creating a copy of a variable (via private/firstprivate/.... clause).
3) Only those constructs which will be outlined like parallel and not those which are inlined like master/workshareloop.

Reply:
The symbolMap scope should more or less match the semantics scope. Any OpenMP region that has its own semantics scope should also have a symbolMap scope, since it may have local symbols that needs local mappings.
I am saying more or less because I think it is also fine if you create symbolMap scopes in cases semantics does not create new scopes.
In lowering, a symbolMap scope is really just a way to create mappings that are only valid in a certain construct and overrides previous symbol mapping in this construct. If no mapping is found in the top level symbolMap scope, the symbolMap scope stack is visited until a match is found, so you adding an symbolMap scope without adding any mappings to it is functionally equivalent to not pushing a new scope (non-functionally, we may lose a bit time looking for symbols in empty scopes, so we should avoid to abusively push scopes if that is not needed, but if it makes the logic a lot easier, that might just be OK).

To match you points:
1. Why not, I'll leave it up to you to judge based on my details above
2. For those, I think you must yes
3. I think it is more a matter of whether you need to define new symbol mappings for a construct. If you only need this for outlined construct, then yes, but if they are local OpenMP symbols in inlined construct, then no.